crates/node/src/lib.rs

```diff
@@ -1042,15 +1082,24 @@ impl Node {
             },
         )?;
 
+        let identity_key =
+            calimero_store::key::ContextIdentity::new(context.id, transaction.executor_public_key);
+        let identity = calimero_primitives::identity::ContextIdentity {
+            public_key: transaction.executor_public_key,
+            private_key: None, // We don't have the private key here
+        };
+        handle.put(&identity_key, &identity)?;
```

Comment:
this isn't meant to happen here, it should happen at `add_context` and `join_context`.

Reply:
I've updated these, but note, although I initially added the context identity to `join_context()`, I removed it again as it felt wrong.